docs/changes/cosmology/12175.api.rst

```diff
@@ -0,0 +1 @@
+The non-public function ``inf_like`` is removed.
```

Comment:
Should there be a change log entry if the public API has not changed?

Reply:
Debatably.
In v4.3- this function lived directly in ``cosmology/core.py`` along with all the publicly scoped functions. Only recently was it moved to ``utils.py`` to make it more clear that this is not publicly scoped.
Additionally it isn't prefixed with an underscore, so I think it's easily mistaken for public API.

Reply:
Better safe than sorry, and including a changelog for an edge-case never hurt anyone.

Reply:
Removing it certainly requires a change log entry then, but it might be better to be more explicit about what exactly has been removed:
```suggestion
The function ``astropy.cosmology.utils.inf_like()`` is removed.
```

Reply:
SGTM. Thanks for the suggestion.